src/features/events.rs

```diff
@@ -3,74 +3,93 @@ use crate::{
   database::{NewEvent, INSTANCE},
 };
 use chrono::{prelude::*, Duration};
-use chrono_tz::Europe::Paris;
+use chrono_tz::{Europe::Paris, Tz};
+use log::info;
 use procedural_macros::command;
-use regex::Regex;
+use regex::{Captures, Regex};
 use serenity::{
   http,
-  model::id::{ChannelId, UserId},
-  prelude::Mentionable,
+  model::{
+    id::{ChannelId, UserId},
+    prelude::{Message, Reaction},
+  },
+  prelude::{Context, Mentionable},
 };
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 use std::{thread, time};
 
+#[derive(Clone)]
+struct RemindMeSnooze {
+  pub author: u64,
+  pub content: String,
+  pub channel: u64,
+  pub message_id: u64,
+}
+
 lazy_static! {
   static ref TIME_INPUT_REGEX: Regex = Regex::new(
     r#"^([0-9]{1,4})((m(inutes?)?)|(h(ours?)?)|(d(ays?)?(([0-9]{2})[:h]([0-9]{2})?)?))$"#
   )
   .expect("unable to create regex");
+  static ref SNOOZE: Mutex<Vec<RemindMeSnooze>> = Mutex::new(Vec::new());
+  static ref SNOOZE_MESSAGE: Mutex<Vec<Message>> = Mutex::new(Vec::new());
```

Comment:
You could do a tupple of the 2 things you want to store instead of iterating 2 times

Also it fixes the problem that this code has no guarantees that It will find the correct message:
```rs
      list_snooze_message.iter().cloned().for_each(|m| {
        let http = ctx.http.clone();
        if m.channel_id == channel {
```

Reply:
I can not regroup both variable in a tuple beacause their index is independant so I cannot do a Vec<(RemindMeSnooze, Message)>
In a second time if i do Mutex<(Vec<RemindMeSnooze, Vec<Message>>)> i cannot push the message to delete because it is the result of asynchrone call and cannot be pushed to mutex
```
future cannot be sent between threads safely
within `impl futures::Future<Output = ()>`, the trait `std::marker::Send` is not implemented for `std::sync::MutexGuard<'_, std::vec::Vec<features::events::RemindMeSnooze>>`
required for the cast from `impl futures::Future<Output = ()>` to the object type `dyn futures::Future<Output = ()> + std::marker::Send`rustc
events.rs(184, 9): future is not `Send` as this value is used across an await
```

Reply:
I am not sure but with what I am reading you actually have 1 Snooze = 1 Message

And for the issue with the Async call this should not be an issue, you probably have to change the way you do things, but you can most probably make it work